src-built-in/components/appLauncher2/src/components/FolderActionsMenu.jsx

```diff
@@ -0,0 +1,76 @@
+/*!
+* Copyright 2018 by ChartIQ, Inc.
+* All rights reserved.
+*
+*/
+
+import React from 'react'
+import storeActions from '../stores/StoreActions'
+
+class FolderActionsMenu extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			isVisible: false
+		};
+		this.bindCorrectContext();
+	}
+	bindCorrectContext() {
+		this.toggleMenu = this.toggleMenu.bind(this);
+		this.renameFolder = this.renameFolder.bind(this);
+		this.launchAll = this.launchAll.bind(this);
+		this.deleteFolder = this.deleteFolder.bind(this);
+	}
+	toggleMenu() {
+		this.setState({
+			isVisible: !this.state.isVisible
+		});
+	}
+	renameFolder() {
+		this.props.renameFolder(this.props.folder.name);
+	}
+	async launchAll() {
```

Comment:
Don't need async here

Reply:
@rcsharp5 I'm not sure why you think there is no need for it there. Well I see @clausjs  is using await storeActions.getFolders() so definitely he needs async there. unless he passes and gets the folders from props or get them in componentWill/DidMount.